### PR TITLE
TGUI Windows no longer auto update by default.

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

TGUI windows no longer auto update by default.
Since so many UIs are being converted to TGUI, ui updates are a significant amount of stuff being done on the server.
I couldn't be bothered to make the UIs perfect, and only update when necessary on all of them, but I have done it on some/most UIs and the support for it is there.

Adds in the ui_update proc, which updates the UI to all viewers on the next TGUI process. Spamming ui_acts won't spam updates, it will just call update on the next normal update.

## Why It's Good For The Game

Should help reduce lag slightly, and adds in support for making future UIs better, or current UIs better in the future.
UIs that are pointlessly being updated should be pointlessly updated less.

## How ports will be affected

Any ported UIs from TG will need to set auto updating to TRUE if the UI requires auto updating, if the UI only updates on act, or when something is inserted into the machine then ui_update() can be called when an update is required.

## Changelog
:cl:
refactor: Refactors TGUI window auto updating.
refactor: TGUI Windows no lonegr auto update by default, and instead need ui_update to be called, or to have their autoupdating var set to TRUE by using set_autoupdating(TRUE)
refactor: Spawners menu is now a single instance rather than on every single ghost.
/:cl:

### CODE REVIEW

File: code/game/machinery/doors/windowdoor.dm [hunk 1]
```diff
@@ -59,6 +59,7 @@
 	if(!open())
 		return
 	autoclose = TRUE
+	wires.ui_update()
```
Comment (Sinestia) [inline code/game/machinery/doors/windowdoor.dm]: Errors whenever a windoor does anything.

It doesn't even have a wire datum, the flag up there lies.

**Review by Sinestia — CHANGES_REQUESTED** [review]

File: code/game/machinery/doors/windowdoor.dm [hunk 2]
```diff
@@ -290,6 +291,7 @@
 /obj/machinery/door/window/try_to_activate_door(mob/user)
 	if (..())
 		autoclose = FALSE
+		wires.ui_update()
```
Comment (PowerfulBacon) [inline code/game/machinery/doors/windowdoor.dm]:
```suggestion
```

File: code/game/machinery/doors/windowdoor.dm (hunk 1 above)
Comment (PowerfulBacon) [inline code/game/machinery/doors/windowdoor.dm]:
```suggestion
```

Comment (PowerfulBacon) [reply code/game/machinery/doors/windowdoor.dm]: Good spot

**Review by ike709 — APPROVED** [review]

**Pull request merged**